src/WebComponents/dom.js

```diff
@@ -99,6 +99,10 @@
       return e;
     };
     window.Event.prototype = origEvent.prototype;
```

Comment:
what about `CustomEvent` above? Did you check if prototype.js also changes that without changing the prototype?